docs/resources/Channel.md

```diff
@@ -505,12 +509,25 @@ Metadata about the interaction, including the source of the interaction and rele
 
 | Field               | Type      | Description                                                                                                                             |
 |---------------------|-----------|-----------------------------------------------------------------------------------------------------------------------------------------|
+| type? \*            | integer   | type of reference. Determines how associated data is populated.                                                                                          |
 | message_id?         | snowflake | id of the originating message                                                                                                           |
-| channel_id? \*      | snowflake | id of the originating message's channel                                                                                                 |
+| channel_id? \*\*    | snowflake | id of the originating message's channel                                                                                                 |
 | guild_id?           | snowflake | id of the originating message's guild                                                                                                   |
 | fail_if_not_exists? | boolean   | when sending, whether to error if the referenced message doesn't exist instead of sending as a normal (non-reply) message, default true |
 
-\* `channel_id` is optional when creating a reply, but will always be present when receiving an event/response that includes this data model.
+\* `type` currently defaults to `DEFAULT` in order to match the behavior when message reference documentation was first published. In a future API version this will be changed to be a required field, with no default.
+\*\* `channel_id` is optional when creating a reply, but will always be present when receiving an event/response that includes this data model.
+
+###### Message Reference Types
+
+> warn
+> If the type is not set, assume a `DEFAULT` type.
+
+| Type            | Value | Description                                              |
+|-----------------|-------|----------------------------------------------------------|
+| DEFAULT         | 0     | A standard reference used by replies.                    |
+| FORWARD         | 1     | Reference used to point to a message at a point in time. |
```

Comment:
is `type` included when receiving the message? if so, then is it DEFAULT for all messages other than forwards (crossposts, follows, pins, replies, thread creates, and thread starters)?

Reply:
`type` will be included. However there will be a period in which old instance will not have this field set, in which case it should be assumed to have `0` set.

>  if so, then is it DEFAULT for all messages other than forwards (crossposts, follows, pins, replies, thread creates, and thread starters)?

I think there's some confusion here. `type` in this case if the MessageReference Type, not the message content type.

Reply:
yeah, but all of those message types use message references for different purposes, they aren't just used for replies. so it feels weird to use one message reference type for 6 different purposes and one type for one purpose.

Reply:
I see the issue. I thought that was a different section based off `Message`. I'll think of a way to clean this up.

Reply:
I took a quick stab last night: https://github.com/discord/discord-api-docs/pull/6818/commits/40c38a43c4d028a30f421141409afd2481a80428